File: java-sdk/src/main/java/com/wikiglobal/wikibroker/openapi/common/Hash.java
package com.wikiglobal.wikibroker.openapi.common;

import lombok.experimental.UtilityClass;

import javax.crypto.Mac;
import javax.crypto.spec.SecretKeySpec;
import java.security.InvalidKeyException;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

@UtilityClass
public class Hash {
    public byte[] hmacSha256(
        byte[] key,
        byte[] message
    ) throws NoSuchAlgorithmException, InvalidKeyException {
        final var restoreKey = new SecretKeySpec(key, "HmacSHA256");
        final var mac = Mac.getInstance(restoreKey.getAlgorithm());
        mac.init(restoreKey);
        return mac.doFinal(message);
    }

    public byte[] sha256Hash(byte[] message) throws NoSuchAlgorithmException {
        final var md = MessageDigest.getInstance("SHA-256");
        md.update(message);
        return md.digest();
    }
}

File: dotnet-sdk/src/Common/Hash.cs
using System.Security.Cryptography;
using System.Text;

namespace WikiBroker.OpenApi.Sdk.Common;

public static class Hash
{
    public static byte[] HmacSha256(byte[] key, byte[] message)
    {
        using var hmac = new HMACSHA256(key);
        return hmac.ComputeHash(message);
    }

    public static byte[] Sha256Hash(byte[] message)
    {
        return SHA256.HashData(message);
    }
}